config/routes.rb

```diff
@@ -18,8 +18,14 @@
   patch "/documents/:id" => "documents#update", as: :document
   get "/documents/:id" => "documents#show"
   get "/documents/:id/generate-path" => "documents#generate_path", as: :generate_path
-  get "/documents/:id/delete-draft" => "documents#confirm_delete_draft", as: :delete_draft
-  delete "/documents/:id" => "documents#destroy"
+
+  get "/documents/:id/delete-draft" => "documents#confirm_delete_draft", as: :delete_draft, constraints: lambda { |request|
```

Comment:
Not sure about putting this in the routes, it's not something that is usual in GOV.UK applications so might be a bit surprising. We had to do something with routes in `frontend`, which became quite complex and introduced very interesting issues. Additionally, this makes a extra database call for these pages, since the controller will have to do `Document.find_by_param` again.

Reply:
Agreed - I do like that it's separate from the controllers, though. At least for me, it's not something I'm interested in when I'm trying to read the controller class. It feels wrong to raise a 500 when nothing's actually broken, and noting we haven't done this for quite a few other routes, perhaps we can mark this as pain and discuss it as a bigger issue?

Reply:
@kevindew done - and I've put a card on the pain wall.